test/e2e/ml_upgrade_test.go

```diff
@@ -0,0 +1,99 @@
+package e2e
+
+import (
+	"fmt"
+	"io/ioutil"
+	"path/filepath"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/gruntwork-io/terratest/modules/helm"
+	"github.com/gruntwork-io/terratest/modules/k8s"
+	"github.com/gruntwork-io/terratest/modules/random"
+	"github.com/stretchr/testify/assert"
+	"github.com/tidwall/gjson"
+	digestAuth "github.com/xinsnake/go-http-digest-auth-client"
+)
+
+func TestHelmMLupgrade(t *testing.T) {
+	// Path to the helm chart we will test
+	helmChartPath, e := filepath.Abs("../../charts")
+	if e != nil {
+		t.Fatalf(e.Error())
+	}
+	username := "admin"
+	password := "admin"
+
+	namespaceName := "marklogic-" + strings.ToLower(random.UniqueId())
+	kubectlOptions := k8s.NewKubectlOptions("", "", namespaceName)
+	options := &helm.Options{
+		KubectlOptions: kubectlOptions,
+		SetValues: map[string]string{
+			"persistence.enabled": "false",
+			"replicaCount":        "1",
+			"image.repository":    "marklogicdb/marklogic-db",
+			"image.tag":           "latest-10.0",
+			"auth.adminUsername":  username,
+			"auth.adminPassword":  password,
+		},
+	}
+
+	t.Logf("====Creating namespace: " + namespaceName)
+	k8s.CreateNamespace(t, kubectlOptions, namespaceName)
+	defer t.Logf("====Deleting namespace: " + namespaceName)
+	defer k8s.DeleteNamespace(t, kubectlOptions, namespaceName)
+
+	t.Logf("====Installing Helm Chart")
+	releaseName := "test-ml-upgrade"
+	helm.Install(t, options, helmChartPath, releaseName)
+
+	podName := releaseName + "-marklogic-0"
+
+	// wait until second pod is in Ready status
+	k8s.WaitUntilPodAvailable(t, kubectlOptions, podName, 20, 20*time.Second)
+
+	newOptions := &helm.Options{
+		KubectlOptions: kubectlOptions,
+		SetValues: map[string]string{
+			"persistence.enabled":   "false",
+			"image.repository":      "marklogicdb/marklogic-db",
+			"image.tag":             "latest",
```

Comment:
We should use a specific version here, as latest will change as we release new versions.

Reply:
added a specific version for MarkLogic upgrade in upgrade_test.go